src/pybind/mgr/cephadm/schedule.py

```diff
@@ -333,7 +341,37 @@ def expand_candidates(ls: List[DaemonPlacement], num: int) -> List[DaemonPlaceme
         existing = existing_active + existing_standby
 
         # build to_add
-        if not count:
+        if self.spec.placement.labels and self.spec.placement.count_per_label:
+            candidate_host_hostnames = [dd.hostname for dd in others if dd.hostname not in [
```

Comment:
Suggestion: extract this block into a function.

Reply:
To clarify: I mean the entire indented block here that starts at this if statement.